File: backend/src/model/pontoInteresse.js
const { DataTypes } = require('sequelize')

module.exports = (sequelize) => {
    sequelize.define('ponto_interesse',
        {
            nome: {
                type: DataTypes.STRING,
                allowNull: false
            },
            morada: {
                type: DataTypes.STRING,
                allowNull: false
            },
            codigo_postal: {
                type: DataTypes.STRING(8),
                allowNull: false,
                validate: {
                    is: {
                        args: /^[0-9]{4}-[0-9]{3}$/i, // regex codigo postal (4 numeros + hiphen + 3 numeros)
                        msg: 'o valor inserido não corresponde ao padrão xxxx-xxx'
                    } 
                }
            },
            num_telemovel: {
                type: DataTypes.STRING(50),
                allowNull: false,
                validate: {
                    is: {
                        args: /^[0-9]{9}$/i, // regex nº tlm (9 numeros)
                        msg: 'o valor inserido não tem 9 números'
                    } 
                }
            },
            num_pontos: {
                type: DataTypes.INTEGER,
                allowNull: false
            },
            descricao: {
                type: DataTypes.STRING,
                allowNull: false
            },
            validado: {
                type: DataTypes.BOOLEAN,
                allowNull: false,
                default: false
            },
            // para evitar um count desnecessário à bd, sempre que há um scan este nº aumenta
            // para mais info, ver o modelo pontos_ponto_interesse
            count_scans: {
                type: DataTypes.INTEGER,
                allowNull: false,
                default: 0
            },
            // para evitar um avg desnecessário à bd, sempre que há uma aval este nº atualiza
            // para mais info, ver o modelo comentario_avaliacao
            avg_avaliacao: {
                type: DataTypes.DECIMAL,
                allowNull: false,
                default: 0.00
            }
        },
        {
            name: { singular: 'ponto_interesse', plural: 'pontos_interesse' },
            underscored: true, // passa de 'createdAt' para 'created_at'. O postgres agradece :)
            freezeTableName: true, // não faz plurais nas relações com outras tabelas. Os devs agradecem :D
            paranoid: true, // na prática, faz com que os records não sejam eliminados, mas sim escondidos (soft-delete) 
            timestamps: true, // created_at, updated_at, e deleted_at
        }
    )
}

File: app/app/src/main/java/pi4/main/PontoInteresse.kt
package pi4.main

class PontoInteresse(
    image_url:String,
    nome:String,
    morada: String,
    descricao: String,
    tipo_interesse:String,
    freguesia_municipio:String,
    num_pontos:String,
    avg_avaliacao:Float,
    count_scans: Int,
    agente_turistico: String
) {
    var image_url:String
    var nome:String
    var morada: String
    var descricao: String
    var tipo_interesse:String
    var freguesia_municipio:String
    var num_pontos:String
    var avg_avaliacao:Float
    var count_scans:Int
    var agente_turistico: String

    init {
        this.image_url = image_url
        this.nome = nome
        this.morada = morada
        this.descricao = descricao
        this.tipo_interesse = tipo_interesse
        this.freguesia_municipio = freguesia_municipio
        this.num_pontos = num_pontos
        this.avg_avaliacao = avg_avaliacao
        this.count_scans = count_scans
        this.agente_turistico = agente_turistico
    }
}